cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityApplier.java

```diff
@@ -378,7 +378,9 @@ String getCurrentVersion() {
   private boolean shouldDeployApp(ApplicationId applicationId, SystemApplication application) throws Exception {
     ApplicationDetail currAppDetail;
     try {
-      currAppDetail = applicationLifecycleService.getAppDetail(applicationId);
```

Comment:
Not sure if the version is always `-SNAPSHOT` for system apps? Any insights?@albertshau

Reply:
This class is responsible for making sure system apps like wrangler and pipeline studio always exist. So it should always get the latest here, and we should change the ApplicationId argument to an ApplicationReference.